nixos/modules/services/backup/mysql-backup.nix

```diff
@@ -8,6 +8,17 @@ let
   cfg = config.services.mysqlBackup;
   defaultUser = "mysqlbackup";
 
+  dumpBinary =
+    if
+      (
+        lib.getName config.services.mysql.package == lib.getName pkgs.mariadb
+        && lib.versionAtLeast config.services.mysql.package.version "11.0.0"
+      )
+    then
+      "${config.services.mysql.package}/bin/mariadb-dump"
+    else
+      "${config.services.mysql.package}/bin/mysqldump";
+
```

Comment:
Not sure about this change, why is this necessary?

Reply:
@infinisil the mariadb team encourages to use mariadb-dump and only has the old mysqldump name as fallback as symlinc.

this is all fine and works, it just states in there logs as warning that it uses an fallback method witch might at some point be removed.

we could either say that the mariadb packager must take care of it and if it is upstream removed readd it ... or I could remove the warning and handle the case - witch i did with this

Reply:
Ah makes sense, could you add this as a code comment? Also explaining why specifically 11.0.0 (I assume that's probably due to that version introducing the warning).

Reply:
https://mariadb.com/kb/en/mysqldump -> to be specific 11.0.1 officially had depricated the symlink but 11.0.0 is easyer to remember and compare ... and i like it more semversion compliant ...

Reply:
added 0932d53270d1417cf2b7012db38defae169f5faa - hope this works :)